src/brs/BurstGUI.java

```diff
@@ -0,0 +1,334 @@
+        try {
+            Burst.main(args);
+            try {
+                if (Burst.getPropertyService().getBoolean(Props.DEV_TESTNET)) {
+                    onTestNetEnabled();
+                }
+            } catch (Throwable t) {
+                LOGGER.error("Could not determine if running in testnet mode", t);
+            }
+        } catch (Throwable t) {
+            if (!(t instanceof SecurityException)) {
+                LOGGER.error("BurstGUI caught exception starting BRS", t);
+                showMessage("BurstGUI caught exception starting BRS");
+                onBrsStopped();
+            }
+        }
+    }
+
+    private static void onTestNetEnabled() {
+        stage.setTitle(stage.getTitle() + " (TESTNET)");
+        trayIcon.setToolTip(trayIcon.getToolTip() + " (TESTNET)");
+    }
+
+    private static void onBrsStopped() {
+        stage.setTitle(stage.getTitle() + " (STOPPED)");
+        trayIcon.setToolTip(trayIcon.getToolTip() + " (STOPPED)");
+    }
+
+    private static void sendJavaOutputToTextArea(TextArea textArea) {
+        System.setOut(new PrintStream(new TextAreaOutputStream(textArea, System.out)));
+        System.setErr(new PrintStream(new TextAreaOutputStream(textArea, System.err)));
+    }
+
+    private static void showMessage(String message) {
+        Platform.runLater(() -> {
+            System.out.println("Showing message: " + message);
+            Dialog dialog = new Alert(Alert.AlertType.ERROR, message, ButtonType.OK);
+            dialog.setGraphic(null);
+            dialog.setHeaderText(null);
+            dialog.setTitle("BRS Message");
+            dialog.show();
+        });
+    }
+
+    private static class TextAreaOutputStream extends OutputStream {
+        private final TextArea textArea;
+        private final PrintStream actualOutput;
+
+        private TextAreaOutputStream(TextArea textArea, PrintStream actualOutput) {
+            this.textArea = textArea;
+            this.actualOutput = actualOutput;
+        }
+
+        @Override
+        public void write(int b) {
+            writeString(new String(new byte[]{(byte)b}));
+        }
+
+        @Override
+        public void write(byte[] b) {
+            writeString(new String(b));
+        }
+
+        private void writeString(String string) {
+            actualOutput.print(string);
+            if (textArea != null) {
+                Platform.runLater(() -> textArea.appendText(string));
+            }
+        }
+    }
+
+    private static class BurstGUISecurityManager extends SecurityManager {
+
+        @Override
+        public void checkExit(int status) {
+            if (!userClosed) {
+                LOGGER.error("BRS Quit unexpectedly! Exit code " + String.valueOf(status));
+                showMessage("BRS Quit unexpectedly! Exit code " + String.valueOf(status));
+                onBrsStopped();
+                throw new SecurityException();
+            }
+        }
+
+        @Override
```

Comment:
I think you intend to overwrite the default permissions, right? Actually you're removing them here. If it's intentionally then it's ok, otherwise it seems like auto-generated overrides, which weren't cleaned up.

Reply:
no, it's intentional. if you never call setSecurityManager the SecurityManager is null so all of these functions are effectively empty but the default SecurityManager class has implementations of them that run all sorts of checks that we don't want, so this is intentional